pkg/api/defaults-custom-cloud-profile.go

```diff
@@ -93,100 +110,114 @@ func (cs *ContainerService) SetCustomCloudProfileEnvironment() error {
+				env.KeyVaultDNSSuffix = fmt.Sprintf("vault.%s.%s", cs.Location, azsFQDNSuffix)
+			} else if (env.ResourceManagerVMDNSSuffix == "" || env.StorageEndpointSuffix == "" || env.KeyVaultDNSSuffix == "") {
+				// Non-AzureStack CustomCloud MUST provide suffixes			
+				return fmt.Errorf("Non-AzureStack CustomCloudProfile MUST provide ResourceManagerVMDNSSuffix, StorageEndpointSuffix, KeyVaultDNSSuffix")
+			}
 		}
 	}
+
 	return nil
 }
 
-// AzureStackCloudSpecParams is the parameters when we set the azure stack cloud spec defaults for ContainerService.
-type AzureStackCloudSpecParams struct {
+// AzureCustomCloudSpecParams is the parameters when we set the azure custom cloud spec defaults for ContainerService.
+type AzureCustomCloudSpecParams struct {
 	IsUpgrade bool
 	IsScale   bool
 }
 
-// SetAzureStackCloudSpec sets the cloud spec for Azure Stack .
-func (p *Properties) SetAzureStackCloudSpec(params AzureStackCloudSpecParams) error {
-	if p.IsAzureStackCloud() {
-		var azureStackCloudSpec AzureEnvironmentSpecConfig
+// SetAzureCustomCloudSpec sets the cloud spec for Custom Cloud .
+func (p *Properties) SetCustomCloudSpec(params AzureCustomCloudSpecParams) error {
+	if p.IsCustomCloudProfile() {
+		var azureCustomCloudSpec AzureEnvironmentSpecConfig
 		switch p.CustomCloudProfile.DependenciesLocation {
-		case AzureStackDependenciesLocationPublic:
-			azureStackCloudSpec = AzureCloudSpecEnvMap[AzurePublicCloud]
-		case AzureStackDependenciesLocationChina:
-			azureStackCloudSpec = AzureCloudSpecEnvMap[AzureChinaCloud]
-		case AzureStackDependenciesLocationGerman:
-			azureStackCloudSpec = AzureCloudSpecEnvMap[AzureGermanCloud]
-		case AzureStackDependenciesLocationUSGovernment:
-			azureStackCloudSpec = AzureCloudSpecEnvMap[AzureUSGovernmentCloud]
+		case AzureCustomCloudDependenciesLocationPublic:
+			azureCustomCloudSpec = AzureCloudSpecEnvMap[AzurePublicCloud]
+		case AzureCustomCloudDependenciesLocationChina:
+			azureCustomCloudSpec = AzureCloudSpecEnvMap[AzureChinaCloud]
+		case AzureCustomCloudDependenciesLocationGerman:
+			azureCustomCloudSpec = AzureCloudSpecEnvMap[AzureGermanCloud]
+		case AzureCustomCloudDependenciesLocationUSGovernment:
+			azureCustomCloudSpec = AzureCloudSpecEnvMap[AzureUSGovernmentCloud]
 		default:
-			azureStackCloudSpec = AzureCloudSpecEnvMap[AzurePublicCloud]
+			azureCustomCloudSpec = AzureCloudSpecEnvMap[AzurePublicCloud]
 		}
 		if p.CustomCloudProfile.Environment == nil || p.CustomCloudProfile.Environment.ResourceManagerVMDNSSuffix == "" {
 			return errors.New("Failed to set Cloud Spec for Azure Stack due to invalid environment")
 		}
 
-		azureStackCloudSpec.EndpointConfig.ResourceManagerVMDNSSuffix = p.CustomCloudProfile.Environment.ResourceManagerVMDNSSuffix
-		azureStackCloudSpec.CloudName = AzureStackCloud
+		azureCustomCloudSpec.EndpointConfig.ResourceManagerVMDNSSuffix = p.CustomCloudProfile.Environment.ResourceManagerVMDNSSuffix
```

Comment:
this logic is only for Azure Stack.

Reply:
I don't think this logic is Azure Stack specific. This logic is simply replacing default VMDNSSuffix with user-provided one.